### PR TITLE
Mixed names for functions

### PR DESCRIPTION
**Pull request by cs18b028**

### CODE REVIEW

File: SemanticAnalysis/parser.y [hunk 1]
```diff
@@ -189,11 +215,19 @@ assign_stmt:                      param assignment
...
                                     printf("assign_stmt\n");
                                     $$ = makeNode(astAssignStmt, NULL, $1, $2, NULL, NULL);
+                                    popV();
+                                    popV();
                                     // sym = find_variable(param->)
                                   }
                                   | arr assignment
                                   {
-                                    $$ = makeNode(astArrayAssignStmt, NULL, $1, $2, NULL, NULL);
+                                    $$ = makeNode(astAssignStmt, NULL, $1, $2, NULL, NULL);
```
Comment (tapish2000) [inline SemanticAnalysis/parser.y]:
```c
$$ = makeNode(astAssignStmt, NULL, $1, $2, NULL, NULL);
```
Why did you change `astArrayAssignStmt` to `astAssignStmt`?

File: SemanticAnalysis/parser.y [hunk 2]
```diff
@@ -339,9 +492,10 @@ param:                            data_type ID
                                     printf("param\n");
                                     
                                     default_value(type);
-                                    sym = makeSymbol($2, type, &value, 0, size, 'v', 1, 0);
+                                    s1 = popV();
+                                    sym = makeSymbol($2, s1->type, &value, s1->size, 'v', 1, 0);
                                     add_variable_to_table(sym);
-                                    $$ = makeNode(astParam, sym, $1, NULL, NULL, NULL);
```
Comment (tapish2000) [inline SemanticAnalysis/parser.y:344]: Are you sure this had to be removed? Don't we need to make it an ast node? If you are sure then some major changes will have to be done in `semantics.c`.

File: SemanticAnalysis/parser.y [hunk 3]
```diff
@@ -562,48 +771,6 @@ void default_value(int type) {
   }
 }
 
-
-/* ------------------- Handling Hash Tables --------------- */
-
-/* ------Return Stack------ */
-
-
-/* ------While Stack------ */
```
Comment (tapish2000) [inline SemanticAnalysis/parser.y:571]: Why did you remove while stack?????

**Review by tapish2000 — CHANGES_REQUESTED** [review]
Please look into all those changes I marked. 

Did you check it by running once?

File: SemanticAnalysis/parser.y (hunk 3 above)
Comment (cs18b028) [reply SemanticAnalysis/parser.y:571]: While stack functions are defined twice(top_while, pop_while,.. all those). So i removed one of those.

File: SemanticAnalysis/parser.y (hunk 1 above)
Comment (cs18b028) [reply SemanticAnalysis/parser.y]: I'll change this back.

File: SemanticAnalysis/parser.y (hunk 2 above)
Comment (cs18b028) [reply SemanticAnalysis/parser.y:344]: I removed this by mistake.

File: SemanticAnalysis/parser.y (hunk 3 above)
Comment (tapish2000) [reply SemanticAnalysis/parser.y:571]: So, you mean it is already there in your latest commit?
If that is the case, then fine. 
I will look once then.

File: SemanticAnalysis/parser.y (hunk 1 above)
Comment (cs18b028) [reply SemanticAnalysis/parser.y]: Resolved this.

File: SemanticAnalysis/parser.y (hunk 2 above)
Comment (cs18b028) [reply SemanticAnalysis/parser.y:344]: Resolved this.

**Review by tapish2000 — APPROVED** [review]
Looks great to me.

**Review by shobhi1310 — APPROVED** [review]
Yes everything is fine. Can be merged!
